### PR TITLE
Update Gradle to 8.3, and node plugin to latest

### LINKED ISSUE #94: Upgrade build to modern version of React, Java and Gradle
see: https://github.com/micronaut-projects/micronaut-starter-ui/pulls/app%2Fdependabot

### PR DESCRIPTION
**Pull request by timyates**
Also updated some dependencies.

I don't think we can move to react 18 without a big upgrade of the material-ui libs.

I'm not 100% this is correct, but it builds, and I can dev locally against a local starter.

Closes #94

### CODE REVIEW

**Review by jeremyg484 — CHANGES_REQUESTED** [review]
Looks good overall and works locally for me, just need to clarify the instructions as noted.

I agree that trying to upgrade to React 18 is probably not worth the trouble at the moment.

File: README.md [hunk 1]
```diff
@@ -7,50 +7,48 @@ This project builds the [Micronaut Launch UI](https://launch.micronaut.io).
...
+10.1.0
+❯ node --version
+v20.8.0
 ```
 
-- Start the dev build of the site
+### Run micronaut-starter locally
 
+```bash
+CORS_ALLOWED_ORIGIN=http://localhost:3000 ./gradlew starter-web-netty:run
```
[the remark this answers is not in the document]
Comment (timyates) [reply README.md]: Pulled the clone bit up here and hopefully made it more explicit 🤞😎

File: README.md [hunk 2]
```diff
@@ -7,50 +7,48 @@ This project builds the [Micronaut Launch UI](https://launch.micronaut.io).
...
-If you want to run against a version of the starter api (Micronaut Starter Api)[https://github.com/micronaut-projects/micronaut-starter]
-
-Outside of this project pull down that repo
+### Run the dev proxy server
 
 ```bash
-git clone git@github.com:micronaut-projects/micronaut-starter.git
-cd micronaut-starter
-export CORS_ALLOWED_ORIGIN=http://localhost:3000
-./gradlew starter-web-netty:run
```
[the remark this answers is not in the document]
Comment (timyates) [reply README.md]: 👍

File: README.md [hunk 3]
```diff
@@ -7,50 +7,48 @@ This project builds the [Micronaut Launch UI](https://launch.micronaut.io).
...
+npm install
+npm run start:starter
 ```
 
-Then start up a version server to provide that instance
+### Run the application locally
 
+```bash
+cd app/launch
+npm run start:local
```
[the remark this answers is not in the document]
Comment (timyates) [reply README.md:52]: Good catch

👍

File: README.md [hunk 4]
```diff
@@ -99,3 +97,45 @@ You can find the version feed file at [./app/launch/public/mn-version-feed.json]
       ...
   },
   ```
+
+---
+## Previous instructions...
```
[the remark this answers is not in the document]
Comment (timyates) [reply README.md]: Gone now... We'll always have git history!

File: build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import com.github.gradle.node.npm.task.NpmTask
```
[the remark this answers is not in the document]
Comment (timyates) [reply build.gradle.kts:1]: I really don't know...  I assume there were reasons at some point...  Maybe gradle handles the npm/node installation requirement better than anything available at the time of this repo's birth 🤔

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (timyates) [reply README.md]: https://github.com/micronaut-projects/micronaut-starter-ui/blob/32c68f2eb8c8648e8119d483d647a62b5ca5d02f/README.md is easier to reads than diffs ;-)

File: build.gradle.kts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jeremyg484) [reply build.gradle.kts:1]: Fair enough. It's simple enough usage that it's not worth changing at the moment.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyg484) [reply README.md]: Yea, I cloned your branch to make sure I could work through the instructions successfully. Looking good now.

**Review by jeremyg484 — APPROVED** [review]

**Review by sdelamo — COMMENTED** [review]
Thanks for this PR @timyates  🙌

Do I need a different `node` version to run it locally?

```
% npm -v                  
10.1.0
sdelamo@Sergios-Mac-Studio micronaut-starter-ui % node -v                 
v18.17.1
sdelamo@Sergios-Mac-Studio micronaut-starter-ui % java --version
openjdk 17.0.8 2023-07-18 LTS
OpenJDK Runtime Environment Corretto-17.0.8.7.1 (build 17.0.8+7-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.8.7.1 (build 17.0.8+7-LTS, mixed mode, sharing)
sdelamo@Sergios-Mac-Studio micronaut-starter-ui % ./gradlew build

> Configure project :
Kotlin DSL property assignment is an incubating feature.

> Task :buildStarter FAILED

> app@0.1.1 build
> react-scripts build

node:internal/modules/cjs/loader:446
      throw err;
      ^

Error: Cannot find module '/Users/sdelamo/github/micronaut-projects/micronaut-starter-ui/app/launch/node_modules/tryer/src/tryer'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:438:19)
    at Module._findPath (node:internal/modules/cjs/loader:680:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1063:27)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:121:18)
    at Object.<anonymous> (/Users/sdelamo/github/micronaut-projects/micronaut-starter-ui/app/launch/node_modules/bfj/src/streamify.js:9:15)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32) {
  code: 'MODULE_NOT_FOUND',
  path: '/Users/sdelamo/github/micronaut-projects/micronaut-starter-ui/app/launch/node_modules/tryer/package.json',
  requestPath: 'tryer'
}

Node.js v18.17.1

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildStarter'.
> Process 'command '/Users/sdelamo/github/micronaut-projects/micronaut-starter-ui/.gradle/nodejs/node-v18.17.1-darwin-arm64/bin/npm'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
```

**Comment by timyates** [comment]
@sdelamo You shouldn't do, as now it downloads and installs a version of node to build the app

So it's being built with

```
> Process 'command '/Users/sdelamo/github/micronaut-projects/micronaut-starter-ui/.gradle/nodejs/node-v18.17.1-darwin-arm64/bin/npm'' finished with non-zero exit value 1
```

Can you try removing your node_modules directory with

```
rm -rf app/launch/node_modules
```

And then try again?

**Comment by sdelamo** [comment]
I tried with a different computer and it loaded 🤷🏻‍♂️

**Pull request merged**